Fixes #26957 - send request id instead session to proxy

lzap opened the pull request:
We send the data to correlate logs but instead request we send session
id. This is confusing and this patch fixes that. Additionally, it also
now sends both so proxy can actually process and log both as both can be
useful when digging in logs.

Can I ask for this one into 1.22? We are targetting improving logging in
1.22 and this is a tiny bug that can make searching for logs confusing.
@tbrisker @ares

lzap committed:
Fixes #26957 - send request id instead session to proxy

tbrisker reviewed (approved):
I assume this also needs some change on the proxy side? Too late for 1.22.0, but this is small enough that i'm willing to pull it into 1.22.1 if the other part is also this small.

tbrisker reviewed (changes requested):
test failures related though

lzap commented:
No changes needed actually for this to work properly, this core change (which I propose for 1.22) fixed the issue - session id is no longer sent as request id. Foreman proxy processes the parameter correctly (it was just bad data coming from core).

Additionally, it now sends both, this will require change in Foreman Proxy (which I am going to work on later today), however this I don't propose into 1.22. That is improvement and it is not what we should be backporting.

Rebased and fixed the tests.

tbrisker reviewed (approved):
Thanks @lzap !

tbrisker commented:
1.22.1- ceb8c2ac2

lzap commented:
Thanks a bunch! I will demo this tomorrow.
